resolving/local-resolver/src/parseBareSpecifier.ts

```diff
@@ -91,14 +93,24 @@ function fromLocal (
     }
   }
 
+  function normalizeRelativeOrAbsolute (relativeTo: string, fromPath: string) {
+    let specPath
+    if (opts.preserveAbsolutePaths && isAbsolute(spec)) {
+      specPath = path.resolve(fromPath)
+    } else {
+      specPath = path.relative(relativeTo, fromPath)
+    }
+    return normalize(specPath)
+  }
+
   injected = protocol === 'file:'
   const dependencyPath = injected
     ? normalize(path.relative(lockfileDir, fetchSpec))
     : normalize(path.resolve(fetchSpec))
```

Comment:
@colinta what about these? These are also added to the lockfile into the resolution object, directory field.

Reply:
Hey I see it now, can update those 👍